src/llvm-late-gc-lowering.cpp

```diff
@@ -424,30 +431,41 @@ static Value *FindBaseValue(const State &S, Value *V, bool UseCache = true) {
             if (!isUnionRep(Operand->getType()))
                 break;
             CurrentV = Operand;
-            continue;
-        } else
+        }
+        else if (auto EEI = dyn_cast<ExtractElementInst>(CurrentV)) {
+            assert(CurrentV->getType()->isPointerTy() && fld_idx == -1);
+            // For now, only support constant index.
+            auto IdxOp = cast<ConstantInt>(EEI->getIndexOperand());
```

Comment:
This value doesn't seem to be guaranteed to be a ConstantInt

Reply:
No it's not. But it covers all the cases I've found in the wield and is strictly more permissive than what was here before.